tools/c7n_azure/c7n_azure/container_host/host.py

```diff
@@ -0,0 +1,379 @@
+        # iterations on other threads
+        policies_copy = self.policies.copy()
+
+        for f in removed_files:
+            path = os.path.join(self.policy_cache, f)
+            self.unload_policy_file(path, policies_copy)
+
+        # Get updated YML files
+        for blob in new_blobs:
+            policy_path = os.path.join(self.policy_cache, blob.name)
+            if os.path.exists(policy_path):
+                self.unload_policy_file(policy_path, policies_copy)
+
+            client.get_blob_to_path(container, blob.name, policy_path)
+            self.load_policy(policy_path, policies_copy)
+            self.blob_cache.update({blob.name: blob.properties.content_settings.content_md5})
+
+        # Assign our copy back over the original
+        self.policies = policies_copy
+
+        if self.require_event_update:
+            self.update_event_subscriptions()
+
+    def load_policy(self, path, policies):
+        """
+        Loads a YAML file and prompts scheduling updates
+        :param path: Path to YAML file on disk
+        :param policies: Dictionary of policies to update
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
+
+                new_policies = PolicyCollection.from_data(policy_config, self.options)
+
+                if new_policies:
+                    for p in new_policies:
+                        log.info("Loading Policy %s from %s" % (p.name, path))
+                        policies.update({p.name: {'policy': p}})
+
+                        # Update periodic and set event update flag
+                        self.update_periodic(p)
+                        if p.data.get('mode', {}).get('events'):
+                            self.require_event_update = True
+
+            except Exception as exc:
+                log.error('Invalid policy file %s %s' % (path, exc))
+
+    def unload_policy_file(self, path, policies):
+        """
+        Unload a policy file that has changed or been removed.
+        Take the copy from disk and pop all policies from dictionary
+        and update scheduled jobs and event registrations.
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                log.warning('Failure loading cached policy for cleanup %s %s' % (path, exc))
+                return
+
+        removed = [policies.pop(p['name']) for p in policy_config.get('policies', [])]
+        log.info('Removing policies %s' % removed)
+
+        # update periodic
+        periodic_names = \
+            [p['name'] for p in policy_config['policies'] if p.get('mode', {}).get('schedule')]
+        periodic_to_remove = \
+            [p for p in periodic_names if p in [j.id for j in self.scheduler.get_jobs()]]
+
+        for name in periodic_to_remove:
+            self.scheduler.remove_job(job_id=name)
```

Comment:
Would be nice to detect policies with the same name here..

Reply:
So the name is the unique ID of the job, so this is always going to remove at most 1 item, which will then get recreated if it was an updated YAML file.  

If/when we do add more warning/intelligence around bad YAML files/duplicates we'd probably want to do it in load policy.